src/lib/viewers/archive/ArchiveExplorer.js

```diff
@@ -132,19 +132,65 @@ class ArchiveExplorer extends React.Component {
      */
     handleClickFullPath = fullPath => this.setState({ fullPath });
 
+    /**
+     * Handle search input, update view state
+     *
+     * @param {string} query - raw query string in the search bar
+     * @return {void}
+     */
+    search = query => {
+        const trimmedQuery = query.trim();
+
+        if (!query) {
+            this.setState({
+                searchQuery: query,
+                view: VIEW_FOLDER,
+            });
+            return;
```

Comment:
We should avoid multiple returns as a method of flow control. We should build up the state object and call `this.setState` once at the bottom of the method.